master/buildbot/status/web/jsontestresults.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+import json
+from twisted.python import log
+from buildbot.status.web.base import HtmlResource, path_to_builder, path_to_builders, path_to_codebases, path_to_build
+
+
+class JSONTestResource(HtmlResource):
+    def __init__(self, log, step_status):
+        HtmlResource.__init__(self)
+        self.log = log
+        self.step_status = step_status
+
+    def etree_to_dict(self, t):
+        d = {t.tag: map(self.etree_to_dict, list(t))}
+        d.update((k, v) for k, v in t.attrib.iteritems())
+        d['text'] = t.text
+        return d
+
+    def suite_command_line(self, suite):
```

Comment:
Could you review the template - this method is used for generating command line to run suite or test locally, that logic from the Yan's page - I supposed he need it